core/sysparam.c

```diff
@@ -295,17 +279,39 @@ static sysparam_status_t _find_entry(struct sysparam_context *ctx, uint16_t matc
 }
 
 /** Read the payload from the current entry pointed to by `ctx` */
+
 static inline sysparam_status_t _read_payload(struct sysparam_context *ctx, uint8_t *buffer, size_t buffer_size) {
-    debug(3, "read payload (%d) @ 0x%08x", min(buffer_size, ctx->entry.len), ctx->addr);
-    CHECK_FLASH_OP(sdk_spi_flash_read(ctx->addr + ENTRY_HEADER_SIZE, (void*) buffer, min(buffer_size, ctx->entry.len)));
+    uint32_t addr = ctx->addr + ENTRY_HEADER_SIZE;
+    size_t size = min(buffer_size, ctx->entry.len);
+    debug(3, "read payload (%d) @ 0x%08x", size, addr);
+
+    CHECK_FLASH_OP(spiflash_read(addr, buffer, buffer_size));
+
+    return SYSPARAM_OK;
+}
+
+static inline sysparam_status_t _compare_payload(struct sysparam_context *ctx, uint8_t *value, size_t size) {
+    debug(3, "compare payload (%d) @ 0x%08x", size, ctx->addr);
+    if (ctx->entry.len != size) return SYSPARAM_NOTFOUND;
+    uint32_t *bounce[BOUNCE_BUFFER_WORDS];
```

Comment:
This seems like a typo. The asterisk is redundant. Luckily pointer and uint32_t are the same size, so it worked without errors.